lib/driver.rb

```diff
@@ -0,0 +1,54 @@
+require_relative 'csv_record'
+
+module RideShare
+  class Driver < CsvRecord
+    attr_reader :name, :vin, :status, :trips
+    attr_writer :status
+
+    def initialize(id:, name:, vin:, status: :AVAILABLE, trips: nil)
+      super(id)
+
+			raise ArgumentError.new("Driver status is not valid") unless %i[AVAILABLE UNAVAILABLE].include?(status)
+      raise ArgumentError.new("VIN must be at least 17 characters") unless vin.length == 17
+      raise ArgumentError.new("Invalid driver ID") unless id > 0
+
+      @name = name
+			@vin = vin
+			@status = status
+      @trips = trips || []
+    end
+
+    def add_trip(trip)
+      @trips << trip
+    end
+
+    def start_trip(trip)
+      @status = :UNAVAILABLE
+      add_trip(trip)
+    end
+
+    def average_rating
+			ratings = []
+			@trips.each do |trip|
+				ratings << trip.rating unless trip.rating.nil?
+			end
+			return @trips.length == 0 ? 0 : (ratings.sum / ratings.length).to_f.round(1)
+	  end
```

Comment:
Just note that you can simplify this type of thing with some enumerables.  Also indentation is a little weird.

```suggestion
      completed_trips = trips.select { |trip| !trip.rating.nil? }
      return 0 if completed_trips.length == 0
      
      total_rating = completed_trips.reduce(0) { |sum, trip| sum + trip.rating }

      return total_rating / completed_trips.length.to_f
```